test/unit/datasets/test_datasets.py

```diff
@@ -38,3 +40,26 @@ def test_dataset_as_X_y(self, as_frame, fetch_function):
         assert isinstance(X, pd.DataFrame if as_frame else np.ndarray)
         assert y is not None
         assert isinstance(y, pd.Series if as_frame else np.ndarray)
+
+    def test_synthetic_datasets(self):
+        """Ensure that dataset creation is deterministic."""
+        rng = np.random.RandomState(12345)
+        X, y, gender = make_synthetic_dataset(random_state=rng)
+
+        X_train, _, y_train, _, gender_train, _ = train_test_split(
+            X, y, gender, test_size=0.3, random_state=rng
+        )
+
+        assert np.sum(X_train[0] < 0) == 8
+        assert np.sum(y_train) == 699
+
+        counts = {'Man': 0, 'Woman': 0, 'Other': 0, 'Unspecified': 0}
+        for k in gender_train:
+            counts[k] += 1
+        expected_counts = {
```

Comment:
This is great! However, can we also check for the number of samples with 0/1? I would hope that there's some significant difference between at least 2 of the groups because that's kind of the point 🙂 
Taking a step back, I don't think we need the train/test split here, because we're not training anything. We're just asserting properties of the utility function you created while assuming that scikit-learn tests `train_test_split` thoroughly. Wdyt?

Reply:
I had included the `train_test_split` because I wanted to test the interaction between the two RNGs. Now that it's shared, there's no benefit, so I've removed it.

> I would hope that there's some significant difference between at least 2 of the groups because that's kind of the point

Of course :) Right now, because the feature is randomly assigned to the samples, we're getting pretty even distribution. I'll add in a utility function to help assign things unfairly.

Reply:
I've made a simple attempt at this in 0d2cda7. Let me know what you think.

Reply:
Good idea! I added a comment separately, and I'm resolving this one. https://github.com/fairlearn/fairlearn/pull/907#discussion_r671991344